mesonpy/__init__.py

```diff
@@ -547,6 +576,62 @@ def build(self, directory: Path) -> pathlib.Path:
 
         return wheel_file
 
+    def build_editable(self, directory: Path, verbose: bool = False) -> pathlib.Path:
+        self._project.build()  # ensure project is built
+
+        wheel_file = pathlib.Path(directory, f'{self.name}.whl')
+
+        install_path = self._source_dir / '.mesonpy' / 'editable' / 'install'
+        rebuild_commands = self._project.build_commands(install_path)
+
+        import_paths = set()
+        for name, raw_path in mesonpy._introspection.SYSCONFIG_PATHS.items():
+            if name not in ('purelib', 'platlib'):
+                continue
+            path = pathlib.Path(raw_path)
+            import_paths.add(install_path / path.relative_to(path.anchor))
+
+        install_path.mkdir(parents=True, exist_ok=True)
+
+        with mesonpy._wheelfile.WheelFile(wheel_file, 'w') as whl:
+            self._wheel_write_metadata(whl)
+            whl.writestr(
+                f'{self.distinfo_dir}/direct_url.json.py',
+                self._source_dir.as_uri().encode(),
```

Comment:
This does not seem all that happy for me, when testing this PR with SciPy on Linux:
```
Successfully built scipy
Installing collected packages: scipy
  *** Error compiling '/home/rgommers/mambaforge/envs/scipy-dev/lib/python3.10/site-packages/scipy-1.11.0.dev0.dist-info/direct_url.json.py'...
    File "/home/rgommers/mambaforge/envs/scipy-dev/lib/python3.10/site-packages/scipy-1.11.0.dev0.dist-info/direct_url.json.py", line 1
      file:///home/rgommers/code/scipy
           ^^
  SyntaxError: invalid syntax
```